launcher/Application.cpp

```diff
@@ -988,6 +993,36 @@ Application::Application(int& argc, char** argv) : QApplication(argc, argv)
         }
     }
 
+#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD)
+
+    // notify user if /tmp is mounted with `noexec` (#1693)
+    {
+        FILE* description = setmntent(MOUNTED, "r");
+        mntent* info = nullptr;
+
+        while ((info = getmntent(description)) != nullptr) {
+            std::string_view directory = info->mnt_dir;
+            std::string_view options = info->mnt_opts;
+
+            if (directory == "/tmp" && options.rfind("noexec") != std::string_view::npos) {
```

Comment:
I wonder if we could use Qt APIs to check if the path is executable in general.

I.e. `QFileInfo("/tmp").isExecutable()`

Reply:
I've tried searching through QT Api to see if there would be anything suitable, but no luck.
Even `QStorageInfo` class doesn't provide such information.

Reply:
I understand that maybe QT doesn't have a straightforward way of doing it, but the question remains: 
can't you just look at the folder if it has the executable permission?


> I.e. `QFileInfo("/tmp").isExecutable()`

Additionally, I did see that the C functions that you use are not 100% the same on all platforms(there are cases where the signature differs).

Reply:
And directory permission don't depend on mount options, unfortunately.

Reply:
> Additionally, I did see that the C functions that you use are not 100% the same on all platforms(there are cases where the signature differs).

Oh yeah, I didn't even notice. OpenBSD has a completely different interface. I'll try fix it today.

Reply:
In that case using `statvfs()` might be a better choice here. See https://www.man7.org/linux/man-pages/man3/statvfs.3.html

That way we don't have to deal with any string operations and can just use `f_flag & ST_NOEXEC` as the condition

Reply:
Thanks, `statvfs` indeed seems like an overall better option. It is also consistent across Linux, FreeBSD and OpenBSD!